Source/WebCore/css/StyleProperties.cpp

```diff
@@ -647,20 +593,10 @@ String StyleProperties::fontSynthesisValue() const
 
 String StyleProperties::get2Values(const StylePropertyShorthand& shorthand) const
 {
-    // Assume the properties are in the usual order start, end.
     ASSERT(shorthand.length() == 2);
-    int startValueIndex = findPropertyIndex(shorthand.properties()[0]);
-    int endValueIndex = findPropertyIndex(shorthand.properties()[1]);
-
-    if (startValueIndex == -1 || endValueIndex == -1)
-        return { };
 
-    auto start = propertyAt(startValueIndex);
-    auto end = propertyAt(endValueIndex);
-
-    // All 2 properties must be specified.
-    if (!start.value() || !end.value())
-        return { };
+    auto start = propertyAt(findPropertyIndex(shorthand.properties()[0]));
+    auto end = propertyAt(findPropertyIndex(shorthand.properties()[1]));
```

Comment:
`end` could just use `getPropertyCSSValue`.

`start` has an `isImplicit` check which doesn't seem to make much sense, maybe it could use `getPropertyCSSValue` too.

Reply:
The `isImplicit` check is going away soon.

I think I‘ll take a pass through tightening up the code again once I finish removing implicit entirely.